server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -829,6 +829,12 @@ private static Set<String> getAllFields(Map<String, Object> input, String prefix
      * @param handler handles the result or failure
      */
     public void executePipeline(Pipeline pipeline, BiConsumer<IngestDocument, Exception> handler) {
+        // shortcut if the pipeline is empty
+        if (pipeline.getProcessors().isEmpty()) {
+            handler.accept(this, null);
+            return;
```

Comment:
Is this going to potentially confuse people who are looking at ingest metrics? I would think it would be a pretty rare case -- how much does this optimization save us? Is it worth the potential future "why are my metrics wrong?" support tickets?

Reply:
What would be wrong about the metrics?

Reply:
Also note that this is the same logic as before, it's just in a slightly different place.

Reply:
Hmm I don't know what would be wrong about the metrics -- I thought I had traced where this would impact them last week, but now I have no idea what I was seeing.

Reply:
No worries, it's certainly a valid question to ask and we are indeed in a maze of twisty passages all alike. 😄